crates/circuit/src/operations.rs

```diff
@@ -2792,3 +2805,101 @@ impl Operation for PyOperation {
         })
     }
 }
+
+#[derive(Clone, Debug)]
+pub enum ArrayType {
+    NDArray(Array2<Complex64>),
+    OneQ(Matrix2<Complex64>),
+    TwoQ(Matrix4<Complex64>),
+}
+
+/// This class is used to wrap a Python side Operation that is not in the standard library
+#[derive(Clone, Debug)]
+// We bit-pack pointers to this, so having a known alignment even on 32-bit systems is good.
```

Comment:
Since we don't support 32-bit anymore, should we adjust the comment?

Reply:
Oops this is a copy and paste error, I'm not sure how this ended up here. I'll clean this up.

Reply:
Fixed in: https://github.com/Qiskit/qiskit/pull/13759/commits/66d397d35aca15d35536971e309db52101656f64